### ISSUE TITLE
x86: issues with movq

### ISSUE THREAD

**Issue by LRGH** [bdcht/amoco#42]
``` python
>>> i = cpu_x86.disassemble('\x66\x0F\xD6\x4D\xF0')
>>> str(i)
'movq        qword ptr [ebp-16], xmm1'
>>> i(mapper())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "amoco/arch/core.py", line 67, in __call__
    i_xxx(self,map)
  File "amoco/arch/x86/asm.py", line 1246, in i_MOVQ
    fmap[op1] = op2.zeroextend(op1.size)
  File "amoco/cas/mapper.py", line 204, in __setitem__
    raise ValueError('size mismatch')
ValueError: size mismatch
```

and also

``` python
>>> str(cpu_x86.disassemble('\xF3\x0F\x7E\x55\xF0'))
'movq        mm2, qword ptr [ebp-16]'
```

where it should be `'movq xmm2, qword ptr [ebp-16]'`

and finally

``` python
>>> i = cpu_x86.disassemble('\xF3\x0F\x7E\xC1')
>>> str(i)
'movq        xmm0, xmm1'
>>> m=mapper()
>>> i(m)
>>> print(str(m))
eip <- { | [0:32]->(eip+0x4) | }
xmm0 <- { | [0:128]->xmm1 | }
```

which should only copy the low quadword of `%xmm1` and clear the high quadword of `%xmm0`

**Comment by LRGH** [comment]
Proposed patch:

``` python
--- a/amoco/arch/x86/asm.py
+++ b/amoco/arch/x86/asm.py
@@ -1243,7 +1246,7 @@ def i_MOVQ(i,fmap):
   fmap[eip] = fmap[eip]+i.length
   op1 = i.operands[0]
   op2 = fmap(i.operands[1])
-  fmap[op1] = op2.zeroextend(op1.size)
+  fmap[op1] = op2[0:64].zeroextend(op1.size)

 def sse_MOVSD(i,fmap):
   fmap[eip] = fmap[eip]+i.length
--- a/amoco/arch/x86/spec_sse.py
+++ b/amoco/arch/x86/spec_sse.py
@@ -491,8 +491,8 @@ def sse_ss(obj,Mod,REG,RM,data):
 def sse_pd(obj,Mod,REG,RM,data,_op2sz):
     if not check_f3(obj,set_opdsz_128): raise InstructionError(obj)
     op2,data = getModRM(obj,Mod,RM,data)
-    if op2._is_mem: op2.size = _op2sz
     op1 = env.getreg(REG,op2.size)
+    if op2._is_mem: op2.size = _op2sz
     obj.operands = [op1,op2]
     obj.type = type_data_processing

```

**Comment by bdcht** [comment]
fixed by commit 1be41d3
